locations/spiders/goodlifefitness.py

```diff
@@ -1,36 +1,37 @@
+import re
+
 import scrapy
+from scrapy.spiders import SitemapSpider
 
-from locations.items import GeojsonPointItem
+from locations.dict_parser import DictParser
+from locations.hours import DAYS, OpeningHours
 
 
-class GoodLifeFitnessSpider(scrapy.Spider):
+class GoodLifeFitnessSpider(SitemapSpider):
     name = "goodlifefitness"
     item_attributes = {"brand": "GoodLife Fitness", "brand_wikidata": "Q3110654"}
-    allowed_domains = ["www.goodlifefitness.com"]
-    start_urls = ("https://www.goodlifefitness.com/locations",)
-
-    def start_requests(self):
-        template = "https://www.goodlifefitness.com/api/club/filterclubsforcitypage"
-
-        headers = {
-            "Accept": "application/json",
-        }
-
-        yield scrapy.http.FormRequest(url=template, method="GET", headers=headers, callback=self.parse)
+    allowed_domains = ["goodlifefitness.com"]
+    sitemap_urls = ["https://www.goodlifefitness.com/sitemap.xml"]
+    sitemap_rules = [(r"/clubs/club.", "parse")]
 
     def parse(self, response):
-        store_data = response.json()
-        for store in store_data:
-            properties = {
-                "ref": store["ClubNo"],
-                "name": store["ClubName"],
-                "addr_full": store["Address1"],
-                "city": store["City"],
-                "state": store["Province"],
-                "postcode": store["PostalCode"],
-                "phone": store["Phone"],
-                "lat": float(store["Lat"]),
-                "lon": float(store["Long"]),
-            }
-
-            yield GeojsonPointItem(**properties)
+        template = "https://www.goodlifefitness.com/content/experience-fragments/goodlife/header/master/jcr:content/root/responsivegrid/header.GetClubsWithDetails.{}.false.true.20221127.json"
+        url_id = re.findall("[0-9]+", response.url)[0]
+        yield scrapy.Request(url=template.format(url_id), callback=self.parse_club)
```

Comment:
Also note, the current process is: get the sitemap -> get the html page -> get the API

We can skip getting the html page if we rewrite the url in `sitemap_filter`.